docs/proposals/pepxxx_wheel_variant_support.md

```diff
@@ -1471,6 +1471,76 @@ Variant marker expressions must be evaluated against the variant properties stor
+
+#### Motivation
+
+Some Python packages with extension modules are built against the ABI of another package and must be used with a
+compatible version of that package at runtime. Traditional Python dependency specifications are insufficient for this
+use case because:
+
+1. The dependency must be satisfied at wheel selection time, not just at install time
+2. Multiple ABI-compatible versions may exist
+3. The package may be built against one of several possible ABI versions
+
+**Example:** vLLM can be built for multiple versions of PyTorch, but once built, it must be used with the exact PyTorch
+ABI it was compiled against. This requirement cannot be expressed with standard dependencies like
+`pip install pytorch==2.8.0 vllm`, forcing projects to distribute source distributions instead of pre-built wheels.
+
+#### Specification
+
+- The namespace identifier must be `abi_dependency`
+- This namespace must not be registered with any variant provider
+- Tools must reject any attempt to register a variant provider using this namespace
+- The feature name represents the distribution name of the package providing the ABI, normalized per the
+  [Binary distribution format](https://packaging.python.org/en/latest/specifications/binary-distribution-format/)
+  specification
+- The feature value represents the version pin
+
+Version values follow a simplified semantic versioning scheme with the following matching rules:
+
+| Variant Property                   | Matching Rule       | Example               |
+|------------------------------------|---------------------|-----------------------|
+| `abi_dependency :: torch :: 2.8.0` | Exact match         | `torch>=2.8.0,<2.8.1` |
+| `abi_dependency :: torch :: 2.9`   | Minor version range | `torch>=2.9,<2.10`    |
+| `abi_dependency :: torch :: 3`     | Major version range | `torch>=3,<4`         |
+
+Only the release segment is permitted, with one to three components.
+
+Per the general rules for variant property matching, multiple variant properties with the feature name express an OR
+relationship, and therefore can be used to indicate wheels compatible with multiple providing package versions, e.g.:
+
+```
+abi_dependency :: torch :: 2.8.0
+abi_dependency :: torch :: 2.9.0
+```
+
+This means the wheel is compatible with both PyTorch 2.8.0 and 2.9.0.
+
+#### Supporting Tools
+
+Tools that implement this feature:
+
+- must evaluate `abi_dependency` constraints during version and wheel resolution
+- must only select wheels whose `abi_dependency` constraints are satisfied by installed (or soon-to-be) packages
+- must interpret version values according to the matching rules specified above
+
+#### Non-Supporting Tools
+
+Tools that do not implement this feature:
+
+- must treat any wheel containing `abi_dependency` variant properties as incompatible
+- must not attempt to install such wheels
+- should provide a clear warning message indicating that the feature is not supported
```

Comment:
The second item follows from the first, we can simplify this to say that `abi_dependency` must be treated as incompatibility and should show a hint for users.